res/controllers/Numark-Party-Mix.scripts.js

```diff
@@ -0,0 +1,440 @@
+        inKey: "parameter1"
+    });
+
+    this.pitch = new components.Pot({
+        inKey: "rate",
+        invert: false
+    });
+
+    this.padSection = new NumarkPartyMix.PadSection(deckNumber);
+
+    this.scratchToggle = new components.Button({
+        midi: [0x90 + channel, 0x07],
+        input: function(channel, control, value, status, group) {
+            if (!this.isPress(channel, control, value)) {
+                return;
+            }
+            deck.scratchModeEnabled = !deck.scratchModeEnabled;
+            if (deck.scratchModeEnabled) {
+                engine.scratchEnable(script.deckFromGroup(group), NumarkPartyMix.jogScratchSensitivity, 33 + 1 / 3, NumarkPartyMix.jogScratchAlpha, NumarkPartyMix.jogScratchBeta, true);
+            } else {
+                engine.scratchDisable(script.deckFromGroup(group));
+            }
+            // change the scratch mode status light
+            this.send(deck.scratchModeEnabled ? this.on : this.off);
+        },
+    });
+
+    this.reconnectComponents(function(component) {
+        if (component.group === undefined) {
+            component.group = this.currentDeck;
+        }
+    });
+};
+
+NumarkPartyMix.Deck.prototype = new components.Deck();
+
+NumarkPartyMix.PadSection = function(deckNumber) {
+    components.ComponentContainer.call(this);
+
+    this.blinkTimer = 0;
+    this.blinkLedState = true;
+
+    // initialize leds
+    var ledOff = components.Button.prototype.off;
+    var ledOn = components.Button.prototype.on;
+    midi.sendShortMsg(0x93 + deckNumber, 0x00, ledOn); // cue
+
+    this.modes = {};
+    this.modes[NumarkPartyMix.PadModeControls.HOTCUE] = new NumarkPartyMix.ModeHotcue(deckNumber);
+    this.modes[NumarkPartyMix.PadModeControls.LOOP] = new NumarkPartyMix.ModeLoop(deckNumber);
+    this.modes[NumarkPartyMix.PadModeControls.SAMPLER] = new NumarkPartyMix.ModeSampler(deckNumber);
+    this.modes[NumarkPartyMix.PadModeControls.EFX] = new NumarkPartyMix.ModeEFX(deckNumber);
+
+    this.modeButtonPress = function(channel, control, value) {
+        this.setMode(channel, control);
+    };
+
+    this.padPress = function(channel, control, value, status, group) {
+        var i = (control - 0x14) % 8;
+        this.currentMode.connections[i].input(channel, control, value, status, group);
+    };
+
+    this.setMode = function(channel, control) {
+        var newMode = this.modes[control];
+        this.currentMode.forEachComponent(function(component) {
+            component.disconnect();
+        });
+
+        newMode.forEachComponent(function(component) {
+            component.connect();
+            component.trigger();
+        });
+
+        this.currentMode = newMode;
+    };
+
+    // start an infinite timer that toggles led state
+    this.blinkLedOn = function(midi1, midi2) {
+        this.blinkLedOff();
+        this.blinkLedState = true;
+        this.blinkTimer = engine.beginTimer(0, function() {
```

Comment:
0 ??
`engine.beginTimer(interval, action, one-shot)`
https://github.com/mixxxdj/mixxx/wiki/Script-Timers

Reply:
the whole timer is unused i removed it.